torch/csrc/api/include/torch/detail/TensorDataContainer.h

```diff
@@ -225,6 +228,11 @@ AT_FORALL_SCALAR_TYPES_AND2(Half, BFloat16, TENSOR)
       fill_tensor(tensor);
       return tensor.to(options.device());
     } else if (is_tensor()) {
+      bool from_complex = tensor_.is_complex();
+      bool to_complex = options.dtype() == at::kComplexFloat || options.dtype() == at::kComplexDouble;
+      if (from_complex && !to_complex) {
+        throw std::runtime_error("can not do torch::tensor(complex, dtype=non-complex) because complex can not be casted to real number without loss of information");
```

Comment:
Is there any reason that we throw std::runtime_error directly here instead of using TORCH_CHECK(false.... or TORCH_INTERNAL_ASSERT(false.....?

Reply:
You are right, I should change to `TORCH_CHECK`

Reply:
fixed